src/Propel/Generator/Reverse/MysqlSchemaParser.php

```diff
@@ -114,6 +114,17 @@ public function parse(Database $database, array $additionalTables = [])
             $this->addPrimaryKey($table);
 
             $this->addTableVendorInfo($table);
+            
+         //Check if the table is the crosstab of a many-to-many relation
+        	$nbPks = count($table->getPrimaryKey());
+            $nbFks = count($table->getForeignKeys());
+            if($nbPks > 1 && $nbFks > 1 && $nbPks == $nbFks) {
```

Comment:
This logic can be simplified, for example:

```suggestion
            if($nbPks === $nbFks && $nbPks > 1) {
```

Reply:
Does this ensure that the table is always what is expected 100% of the time in all possible cases, or does it simply tend to be this way in some specific/common usecases?